kratos/controllers/controller.h

```
@@ -0,0 +1,175 @@
+    KRATOS_CLASS_POINTER_DEFINITION(Controller);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    Controller() noexcept = default;
+
+    /// Destructor.
+    virtual ~Controller() {}
+
+    /// Copy constructor.
+    //TODO: Check. It is required by the registry
+    Controller(Controller const& rOther) {}
+
+    /// Move constructor
+    Controller(Controller&& rOther) noexcept = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    Controller& operator=(Controller const& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method creates an pointer of the controlle
+     * @details We consider as input a Model and a set of Parameters for the sake of generality
+     * @warning Must be overrided in controller implementation
+     * @param rModel The model to be consider
+     * @param ThisParameters The configuration parameters
+     */
+    virtual Controller::Pointer Create(
+        Model& rModel,
+        Parameters ThisParameters) const
+    {
+        KRATOS_ERROR << "Calling base class Create. Please override this method in the corresonding Controller" << std::endl;
+        return nullptr;
+    }
+
+    /**
+     * @brief Checks that input conditions are correct. 
+     */
+    virtual int Check()
+    {
+        return 0;
+    }
+
+    /**
+     * @brief Using input data, returns bool.
+     */
+    virtual bool Evaluate()
+    {
+        KRATOS_ERROR << "Calling base class Evaluate. Please override this method in the corresonding Controller" << std::endl;
+    }
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    virtual Parameters GetDefaultParameters() const
+    {
+        KRATOS_ERROR << "Calling the base Controller class GetDefaultParameters. Please implement the GetDefaultParameters in your derived process class." << std::endl;
```

Not sure if we should throw an error in here. Note that we might want to add some common fields in here in the future (e.g. echo level).

---

hmm ok, returning an empty parameters